### ISSUE TITLE
serverobjects.sql Inserts Class name without " "

### ISSUE THREAD

**Issue by osuapoc** [micovery/extdb-wasteland#2]
Upon using the SQL insert function for server objects, I noticed that the class names for all of the inserted objects within the MySQL DB do not have the encapsulating " " of objects saved during normal server operation.  Upon server startup, these objects are deleted by the server.

**Comment by osuapoc** [comment]
Seems this is also occurring with the servervehicles.sql insert functions as well.  Perchance I've entered something wrong in the cmd when executing, but even when running just the required variable (Path) it still created the insert functions this way.

If it matters, I'm using HeidiSQL to interact with the db.

**Comment by micovery** [comment]
You are right, it's a bug. Thanks for reporting.

**Comment by micovery** [comment]
@AgentRev, what other fields need to be explicitly quoted?

**Comment by AgentRev** [comment]
All strings that are supposed to go in PlayerSave, ServerVehicles, and ServerObjects, must be wrapped in double quotes, also HackType & HackValue in AntihackLog, and ActionType & ActionValue in AdminLog.

Note that all data saved by iniDB is stored within double quotes; the type that each field needs to be converted to after being read is hardcoded in the iniDB mission scripts.

**Comment by micovery** [comment]
@osuapoc I've put a fix on a separate branch master+fix_class_name

Try uninstalling the module:

```sh
  npm uninstall - extdb-wasteland
```

and then install it from the branch here directly in GitHub like this:

```
npm install -g git+https://github.com/micovery/extdb-wasteland.git#master+fix_class_name
```

**Comment by osuapoc** [comment]
Finally got a chance to test this out.  Seems like the classnames converted just fine.  However, I'm getting a few errors in the server RPT and I'm unable to progress past the "Player Info..." stage of load-in.  

Here's my log file:
http://pastebin.com/HfandgNG

This is a modified version of A3Wasteland 1.1.  Based off of TOPArma server's tweaks for flag conquest vs. the normal zones.  The iniDB saves are from a v1.0C release of A3Wasteland.

**Comment by micovery** [comment]
Hmm looking at the log, looks like the mission sent a request without a value for the ```A3W_extDB_MapID``` variable ... not being set.

On your database, what does the ServerMap table contain? Do a select on it, and paste it here.

**Comment by osuapoc** [comment]
Dropped the DB, created a new one.  Was able to load in without any heartache.

So then I only loaded in the objects and vehicles into a fresh database, and I was able to load in, however only the objects persisted.  It seems there was some error in pulling the vehicles back out of the DB.

Here's the snippet from the log for that event:
http://pastebin.com/aFeymb7a

**Comment by micovery** [comment]
Looks like an issue with the vehicle texture that were not migrated correctly. Can you do a test ... start fresh, and save, and reload a vehicle that has some textures, then paste here how the data for the vehicle looks in the database. I probably need to fix the textures field as well.

**Comment by micovery** [comment]
This is how textures get inserted right now when migrated:

```
"[[\"client\\images\\vehicleTextures\\rainbow.jpg\",[0]]]"
```

Which is I am guessing it snot what extDB expects.

**Comment by osuapoc** [comment]
So here's the insert command from the row export:

INSERT INTO `servervehicles` (`ID`, `ServerID`, `MapID`, `CreationDate`, `LastUsed`, `Class`, `Position`, `Direction`, `Velocity`, `Flying`, `Damage`, `Fuel`, `HitPoints`, `Variables`, `Textures`, `Weapons`, `Magazines`, `Items`, `Backpacks`, `TurretMagazines`, `TurretMagazines2`, `TurretMagazines3`, `AmmoCargo`, `FuelCargo`, `RepairCargo`) VALUES (1, 1, 1, '2015-01-24 23:41:30', '2015-01-24 23:41:59', '"O_Quadbike_01_F"', '["14590.892578","16270.603516","1.409822"]', '[[-0.949463,-0.313879,0.000671387],[0.000673536,0.000101597,1]]', '[0,0,0]', 0, 0, 0.998896, '[["HitEngine",0],["HitFuel",0],["HitLFWheel",0],["HitLF2Wheel",0],["HitRFWheel",0],["HitRF2Wheel",0],["HitRGlass",0],["HitLGlass",0],["HitGlass1",0],["HitGlass2",0],["HitGlass3",0],["HitGlass4",0],["HitGlass5",0],["HitGlass6",0],["HitBody",0],["HitLBWheel",0],["HitLMWheel",0],["HitRBWheel",0],["HitRMWheel",0]]', '[["ownerUID","76561197971991469"],["A3W_purchasedVehicle",true]]', '[["client\\images\\vehicleTextures\\confederate.jpg",[0]]]', '[]', '[]', '[]', '[]', '[]', '[]', '[]', 0, 0, 0);

**Comment by osuapoc** [comment]
Result of Import (Textures field):
"[[\"client\\images\\vehicleTextures\\rainbow.jpg\",[0]]]"

From in game save:
[["client\images\vehicleTextures\confederate.jpg",[0]]]

**Comment by micovery** [comment]
Ok, yeah that's an issue. Let me push one more commit with a fix for this in into the fix_class_name branch.

**Comment by osuapoc** [comment]
Like a champ, sir.  Thanks for all of the help!

**Comment by micovery** [comment]
Btw, in order to avoid the back-and-forth ... can you also check these other fields:
```
        JSON.stringify(obj["Hitpoints"]) || "[]",
        JSON.stringify(obj["Variables"]) || "[]",
        JSON.stringify(obj["Textures"]) || "[]",
        JSON.stringify(obj["Weapons"]) || "[]",
        JSON.stringify(obj["Magazines"]) || "[]",
        JSON.stringify(obj["Items"]) || "[]",
        JSON.stringify(obj["Backpacks"]) || "[]",
        JSON.stringify(obj["TurretMagazines"]) || "[]",
        JSON.stringify(obj["TurretMagazines2"]) || "[]",
        JSON.stringify(obj["TurretMagazines3"]) || "[]",
```
That are being converted to string ... to see if they don't match the in-game save.

**Comment by osuapoc** [comment]
The only one that I can actually tell looks weird is the hitpoints field

Here's the data for an ingame save:
[["HitEngine",0],["HitFuel",0],["HitLFWheel",0],["HitLF2Wheel",0],["HitRFWheel",0],["HitRF2Wheel",0],["HitRGlass",0],["HitLGlass",0],["HitGlass1",0],["HitGlass2",0],["HitGlass3",0],["HitGlass4",0],["HitGlass5",0],["HitGlass6",0],["HitBody",0],["HitLBWheel",0],["HitLMWheel",0],["HitRBWheel",0],["HitRMWheel",0]]

Whereas the imported seems to only show:
[]

**Comment by micovery** [comment]
I see ... well, the ```[]``` I've put for the cases where there is no information available, so I assumed extDB would have been fine with an empty array. For the texture problem, It seems the issue is only with the ones that contain back-slashes (like a file path). Could you upload (maybe pastebin) a zip with some vehicle saves that have textures so I can try it out.

**Comment by osuapoc** [comment]
Very possible that it imported that (hitpoints) OK, as it made it deeper into the vehicle data before having a seizure.

I'm not sure how I'd go about uploading a zip file to github, so I dropped it to a pastebin again:
http://pastebin.com/kXrdrw1c

**Comment by micovery** [comment]
Pastebin works fine. I've put a fix for the textures. Re-install the node package again using:

```
npm install -g git+https://github.com/micovery/extdb-wasteland.git#master+fix_class_name
```

And try it out. 

Btw, thanks for the patience with this :bow:  As you can see I am not an extDB user ... so I did not fully test it. You are the first full-tested :smile: .

**Comment by osuapoc** [comment]
Something is unhappy in the install!

Here's the log file from npm when I try running the installer:
http://pastebin.com/WaHZGaK3

**Comment by micovery** [comment]
Hmm that's weird... it says you don't have Git installed. How did you install it last time? Try running the command from the [Git Bash](https://msysgit.github.io) console.

**Comment by osuapoc** [comment]
For some reason I think my path variable likes to mock me, so I often have to re-add stuff to it even though it shows in the system variables.  

Installing through GitBash did the trick though. 

I managed to get a vehicle from the inidb data to the extDB test server with a customer texture and it seems to have loaded in good.  You were indeed correct that the extDB load in did not care about the empty array for Hitpoints field.  It auto-populated that upon spawning of the vehicle.

Good work, sir!  Very helpful!

**Comment by micovery** [comment]
No problem, glad it worked out. I'll merge the fixes from this branch into master. If you notice any other issue, or stuff that is not migrated correctly, do report it. Cheers.

**Comment by osuapoc** [comment]
The only thing to watch is that the Playersave data is pretty wonky yet, I believe.  Though I am much less concerned about that.

**Comment by micovery** [comment]
Closing, branch as been merged into master (656fff706d510b436d3283b7498804c455094df3)
